web-search-engines/langchain4j-web-search-engine-searchapi/src/main/java/dev/langchain4j/web/search/searchapi/SearchApiRequest.java

```diff
@@ -0,0 +1,105 @@
+import static dev.langchain4j.internal.Utils.isNotNullOrBlank;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@Builder
+class SearchApiRequest {
+
+	/************************************************************************************************ 
+	 * Mandatory parameters 
+	 * **********************************************************************************************/
+	
+	/* Search API key. */
+    private String apiKey;
+
+	/* Search engine to retrieve real-time data from. Defaults to "google". */
+    private String engine;
+    
+    /* Search query. */
+    private String q;
+
+    
+    
+    /************************************************************************************************  
+	 * Optional parameters 
+	 * **********************************************************************************************/
+    
+    /* Search device. Valid values: "desktop", "mobile" or "tablet". Defaults to "desktop". */
+    private String device;
+    
+    /** Geographic Location */ 
+    /* Location from where you want the search to originate. Examples "new york" or "london". Valid values see: https://www.searchapi.io/docs/locations-api */
+    private String location;
+    
+    /* Automatically generated from the "location" parameter but can be overridden. "uule" and "location" parameters can't be used together. */
+    private String uule;
+    
+    
+    /** Localization */
+    /* The Google domain of the search. Defaults to "google.com". Full list of supported Google domains, see: https://www.searchapi.io/docs/parameters/google/domain. */
+    private String googleDomain;
+    
+    /* Together hl & gl when concatenated become "en-us".
+    /* Two-letter language code for the search interface. Defaults to "en". Full list of supported "hl" values, see https://www.searchapi.io/docs/parameters/google/hl. */
+    private String hl;
+
+    /* Two-letter country code for the search language. Defaults to "us". Full list of supported "gl" values, see https://www.searchapi.io/docs/parameters/google/gl. */
+    private String gl; 
+
+    
+    
+    /** Filters */
+    /* Safe search. Valid values: "active" or "off". Defaults to "off". */
+    private Boolean safe;
+    
+    /* "Duplicate Content" and "Host Crowding" filters. Defaults to "1" (enabled). To disable, set the value to 0. */
+    private Integer filter;
+    
+    
+    /** Pagination */
+    /* Number of results to display per page. */
+    private Integer num;
+    
+    /* The results page to fetch next. Defaults to 1. */
+    private Integer page;
+    
+    
+    
+    
+    /* Returns a map of key-value pairs representing optional parameters for the search request. */ 
+    public Map<String, Object> getOptionalParameters() {
+    	final Map<String, Object> params = new HashMap<String, Object>();
+    	
+    	if (isNotNullOrBlank(device)) params.put("device", getOrDefault(device, "desktop"));
```

Comment:
hmm, this is strange - `getOrDefault` returns default value if provided value is null, but this cannot be the case because the `isNotNullOrBlank` before. in other words if `device` is null, it won't come to default 'desktop', never

Reply:
and according to the docs, the default value on the server side is already 'desktop':
![image](https://github.com/langchain4j/langchain4j/assets/33568148/465bcb3e-4ebe-440e-9714-205bca337ef7)
so, `getOrDefault` makes no sense on the client side

Reply:
No need to be patronizing in your feedback. 

You could have delivered the same message by merely writing: "the `getOrDefault()` call is not necessary".

Reply:
I've removed it.

Reply:
I've provided my thoughts about the check logic, where you possibly had some another intention while writing the code... 
So, there are still two same checks for "hl" and "gl".

Reply:
Thanks for catching. Fixed.